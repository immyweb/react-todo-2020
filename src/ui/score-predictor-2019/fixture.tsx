import * as React from 'react';

import Select from './select';
import { getBadge } from './utils';

import { FixtureTdTeam, FixtureBadge, FixtureTdVs, FixtureScore } from './styles';

import { ITally, ITeam, IGame, FixtureResultType } from './types';

interface IProps {
  fixture: IGame;
  teams: ITeam[];
  onPredict(tally: ITally[]): void;
}

interface IState {
  team1Score: number;
  team2Score: number;
}

export default class Fixture extends React.Component<IProps, IState> {
  constructor(props: IProps) {
    super(props);
    this.state = {
      team1Score: 0,
      team2Score: 0,
    };
  }

  componentDidMount() {
    const { team1, team2, id, startPrediction } = this.props.fixture;
    this.setState({
      team1Score: startPrediction.team1,
      team2Score: startPrediction.team2,
    });

    const teamOneResultType = this.resultType(startPrediction.team1, startPrediction.team2);
    const teamTwoResultType = this.resultType(startPrediction.team2, startPrediction.team1);

    this.props.onPredict([
      {
        id,
        team: team1,
        resultType: teamOneResultType,
        goals: startPrediction.team1,
        conceded: startPrediction.team2,
      },
      {
        id,
        team: team2,
        resultType: teamTwoResultType,
        goals: startPrediction.team2,
        conceded: startPrediction.team1,
      },
    ]);
  }

  resultType(team1Score: number, team2Score: number) {
    if (team1Score > team2Score) {
      return FixtureResultType.Win;
    }
    if (team1Score < team2Score) {
      return FixtureResultType.Lose;
    }
    return FixtureResultType.Draw;
  }

  checkPlayed() {
    const { team1, team2, id } = this.props.fixture;
    const { team1Score, team2Score } = this.state;

    const teamOneResultType = this.resultType(team1Score, team2Score);
    const teamTwoResultType = this.resultType(team2Score, team1Score);

    this.props.onPredict([
      {
        id,
        team: team1,
        resultType: teamOneResultType,
        goals: team1Score,
        conceded: team2Score,
      },
      {
        id,
        team: team2,
        resultType: teamTwoResultType,
        goals: team2Score,
        conceded: team1Score,
      },
    ]);
  }

  onSelectTeamOne = (value: string) => {
    this.setState({ team1Score: parseInt(value, 10) });
    this.checkPlayed();
  };

  onSelectTeamTwo = (value: string) => {
    this.setState({ team2Score: parseInt(value, 10) });
    this.checkPlayed();
  };

  render() {
    const { fixture, teams } = this.props;

    return (
      <tr>
        <FixtureTdTeam>
          <FixtureBadge src={getBadge(fixture.team1, teams)} alt={fixture.team1} />
          {fixture.team1}
        </FixtureTdTeam>
        <FixtureScore>
          <Select
            team={fixture.team1}
            startPrediction={fixture.startPrediction.team1}
            onSelectTeam={this.onSelectTeamOne}
          />
        </FixtureScore>
        <FixtureTdVs>vs</FixtureTdVs>
        <FixtureScore>
          <Select
            team={fixture.team2}
            startPrediction={fixture.startPrediction.team2}
            onSelectTeam={this.onSelectTeamTwo}
          />
        </FixtureScore>
        <FixtureTdTeam>
          <FixtureBadge src={getBadge(fixture.team2, teams)} alt={fixture.team2} />
          {fixture.team2}
        </FixtureTdTeam>
      </tr>
    );
  }
}
