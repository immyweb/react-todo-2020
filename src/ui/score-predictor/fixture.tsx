import * as React from 'react';
import glamorous from 'glamorous';

import { TdTeam, TrTeam, TdCta, Badge, BadgeLabel } from './styles';
import { colours } from '../../styles/colours';

import { ITally, ITeam, IFixture, FixtureResultType } from './types';

export const getBadge = (teamName: string, teams: ITeam[]): string => {
  const team = teams.find(t => t.team === teamName);
  if (team !== undefined) {
    return team.badge;
  }
  return '';
};

interface IProps {
  fixture: IFixture;
  teams: ITeam[];
  onPredict(tally: ITally[]): void;
  onPlayed(fixtureId: number): void;
  onSlidePlayed(fixtureId: number): void;
}

interface IState {
  played: boolean;
  homeWin: boolean;
  awayWin: boolean;
  draw: boolean;
}

export class Fixture extends React.Component<IProps, IState> {
  constructor(props: IProps) {
    super(props);
    this.state = {
      played: false,
      homeWin: false,
      awayWin: false,
      draw: false,
    };
  }

  onSelection(result: string) {
    if (result === FixtureResultType.Home) {
      this.setState({
        played: true,
        homeWin: true,
        awayWin: false,
        draw: false,
      });
    } else if (result === FixtureResultType.Draw) {
      this.setState({
        played: true,
        homeWin: false,
        awayWin: false,
        draw: true,
      });
    } else {
      this.setState({
        played: true,
        homeWin: false,
        awayWin: true,
        draw: false,
      });
    }
  }

  render() {
    const { fixture, teams, onPredict, onPlayed, onSlidePlayed } = this.props;

    return (
      <TrTeam>
        <TdTeam>
          <Badge
            alt={fixture.team1}
            src={getBadge(fixture.team1, teams)}
            draw={this.state.draw}
            lose={this.state.awayWin}
          />
          <BadgeLabel draw={this.state.draw} lose={this.state.awayWin}>
            {fixture.team1}
          </BadgeLabel>
        </TdTeam>
        <TdCta align={'right'}>
          <Cta
            onClick={() => {
              onPredict([
                {
                  team: fixture.team1,
                  id: fixture.id,
                  resultType: FixtureResultType.Win,
                },
                {
                  team: fixture.team2,
                  id: fixture.id,
                  resultType: FixtureResultType.Lose,
                },
              ]);
              onPlayed(fixture.id);
              this.onSelection(FixtureResultType.Home);
              onSlidePlayed(fixture.id);
            }}
            active={this.state.homeWin}
          >
            W
          </Cta>
        </TdCta>
        <TdCta>
          <Cta
            onClick={() => {
              onPredict([
                {
                  team: fixture.team1,
                  id: fixture.id,
                  resultType: FixtureResultType.Draw,
                },
                {
                  team: fixture.team2,
                  id: fixture.id,
                  resultType: FixtureResultType.Draw,
                },
              ]);
              onPlayed(fixture.id);
              this.onSelection(FixtureResultType.Draw);
              onSlidePlayed(fixture.id);
            }}
            active={this.state.draw}
          >
            D
          </Cta>
        </TdCta>
        <TdCta align={'left'}>
          <Cta
            onClick={() => {
              onPredict([
                {
                  team: fixture.team2,
                  id: fixture.id,
                  resultType: FixtureResultType.Win,
                },
                {
                  team: fixture.team1,
                  id: fixture.id,
                  resultType: FixtureResultType.Lose,
                },
              ]);
              onPlayed(fixture.id);
              this.onSelection(FixtureResultType.Away);
              onSlidePlayed(fixture.id);
            }}
            active={this.state.awayWin}
          >
            W
          </Cta>
        </TdCta>
        <TdTeam>
          <Badge
            alt={fixture.team2}
            src={getBadge(fixture.team2, teams)}
            draw={this.state.draw}
            lose={this.state.homeWin}
          />
          <BadgeLabel draw={this.state.draw} lose={this.state.homeWin}>
            {fixture.team2}
          </BadgeLabel>
        </TdTeam>
      </TrTeam>
    );
  }
}

interface IGProps {
  active?: boolean;
}

export const Cta = glamorous.button<IGProps>(
  {
    display: 'inline-block',
    boxShadow: 'none',
    borderRadius: '50%',
    padding: 0,
    height: 32,
    width: 32,
    fontSize: 14,
    appearance: 'none',
    cursor: 'pointer',
    textAlign: 'center',
    textShadow: 'none',
    textDecoration: 'none',
    WebkitFontSmoothing: 'antialiased',
    background: 'none',
    ':focus': {
      outline: 0,
    },
  },
  ({ active }) => ({
    label: active ? 'slides-cta-active' : 'slides-cta',
    background: active ? colours.$football : colours.$white,
    border: active ? `1px solid ${colours.$football}` : `1px solid ${colours.$darkGrey}`,
    color: active ? colours.$white : colours.$darkGrey,
    fontWeight: active ? 'bold' : 'normal',
  }),
);
