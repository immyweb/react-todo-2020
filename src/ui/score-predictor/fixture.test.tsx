import * as React from 'react';
import { shallow } from 'enzyme';
import { Fixture, getBadge, Cta } from './fixture';

const teams = [
  {
    team: 'Liverpool',
    badge: 'https://www.thesun.co.uk/wp-content/uploads/2016/04/Liverpool.png?strip=all&w=67',
    played: 31,
    win: 18,
    draw: 9,
    lose: 4,
    points: 63,
  },
  {
    team: 'C Palace',
    badge:
      'https://www.thesun.co.uk/wp-content/uploads/2016/04/CPFC-Primary-Badge.png?strip=all&w=72',
    played: 31,
    win: 7,
    draw: 9,
    lose: 15,
    points: 30,
  },
];

function shallowFixture(props?: any) {
  return shallow(
    <Fixture
      fixture={{ id: 0, team1: 'C Palace', team2: 'Liverpool' }}
      teams={teams}
      onPlayed={() => {}}
      onPredict={() => {}}
      onSlidePlayed={() => {}}
      {...props}
    />,
  );
}

describe('<Fixture />', () => {
  it('renders correctly', () => {
    const wrapper = shallowFixture();
    expect(wrapper).toMatchSnapshot();
  });

  it('getBadge renders correct badge for a team', () => {
    const teamBadge = getBadge('Liverpool', teams);
    expect(teamBadge).toEqual(
      'https://www.thesun.co.uk/wp-content/uploads/2016/04/Liverpool.png?strip=all&w=67',
    );
  });

  it('should call onPredict when a match is predicted', () => {
    const fn = jest.fn();
    const wrapper = shallowFixture({ onPredict: fn });
    const btnHome = wrapper.find(Cta).first();
    btnHome.simulate('click');
    expect(fn).toHaveBeenCalledTimes(1);
    fn.mockClear();
    const btnAway = wrapper.find(Cta).last();
    btnAway.simulate('click');
    expect(fn).toHaveBeenCalledTimes(1);
    fn.mockClear();
    const btnDraw = wrapper.find(Cta).at(1);
    btnDraw.simulate('click');
    expect(fn).toHaveBeenCalledTimes(1);
  });

  it('should call onPlayed when a match is predicted', () => {
    const fn = jest.fn();
    const wrapper = shallowFixture({ onPlayed: fn });
    const btnHome = wrapper.find(Cta).first();
    btnHome.simulate('click');
    expect(fn).toHaveBeenCalledTimes(1);
    fn.mockClear();
    const btnAway = wrapper.find(Cta).last();
    btnAway.simulate('click');
    expect(fn).toHaveBeenCalledTimes(1);
    fn.mockClear();
    const btnDraw = wrapper.find(Cta).at(1);
    btnDraw.simulate('click');
    expect(fn).toHaveBeenCalledTimes(1);
  });

  it('should call onSlidePlayed when a match is predicted', () => {
    const fn = jest.fn();
    const wrapper = shallowFixture({ onSlidePlayed: fn });
    const btnHome = wrapper.find(Cta).first();
    btnHome.simulate('click');
    expect(fn).toHaveBeenCalledTimes(1);
    fn.mockClear();
    const btnAway = wrapper.find(Cta).last();
    btnAway.simulate('click');
    expect(fn).toHaveBeenCalledTimes(1);
    fn.mockClear();
    const btnDraw = wrapper.find(Cta).at(1);
    btnDraw.simulate('click');
    expect(fn).toHaveBeenCalledTimes(1);
  });

  it('should set correct state when HOME win is predicted', () => {
    const wrapper = shallowFixture();
    const btnHome = wrapper.find(Cta).first();
    btnHome.simulate('click');
    expect(wrapper.state('played')).toBe(true);
    expect(wrapper.state('homeWin')).toBe(true);
    expect(wrapper.state('awayWin')).toBe(false);
    expect(wrapper.state('draw')).toBe(false);
  });

  it('should set correct state when AWAY win is predicted', () => {
    const wrapper = shallowFixture();
    const btnHome = wrapper.find(Cta).last();
    btnHome.simulate('click');
    expect(wrapper.state('played')).toBe(true);
    expect(wrapper.state('homeWin')).toBe(false);
    expect(wrapper.state('awayWin')).toBe(true);
    expect(wrapper.state('draw')).toBe(false);
  });

  it('should set correct state when DRAW is predicted', () => {
    const wrapper = shallowFixture();
    const btnHome = wrapper.find(Cta).at(1);
    btnHome.simulate('click');
    expect(wrapper.state('played')).toBe(true);
    expect(wrapper.state('homeWin')).toBe(false);
    expect(wrapper.state('awayWin')).toBe(false);
    expect(wrapper.state('draw')).toBe(true);
  });
});
