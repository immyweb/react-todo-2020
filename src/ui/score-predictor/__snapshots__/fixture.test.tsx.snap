// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Fixture /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Fixture
    fixture={
      Object {
        "id": 0,
        "team1": "C Palace",
        "team2": "Liverpool",
      }
    }
    onPlayed={[Function]}
    onPredict={[Function]}
    onSlidePlayed={[Function]}
    teams={
      Array [
        Object {
          "badge": "https://www.thesun.co.uk/wp-content/uploads/2016/04/Liverpool.png?strip=all&w=67",
          "draw": 9,
          "lose": 4,
          "played": 31,
          "points": 63,
          "team": "Liverpool",
          "win": 18,
        },
        Object {
          "badge": "https://www.thesun.co.uk/wp-content/uploads/2016/04/CPFC-Primary-Badge.png?strip=all&w=72",
          "draw": 9,
          "lose": 15,
          "played": 31,
          "points": 30,
          "team": "C Palace",
          "win": 7,
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <glamorous(td)>
          <glamorous(img)
            alt="C Palace"
            draw={false}
            lose={false}
            src="https://www.thesun.co.uk/wp-content/uploads/2016/04/CPFC-Primary-Badge.png?strip=all&w=72"
          />
          <glamorous(span)
            draw={false}
            lose={false}
          >
            C Palace
          </glamorous(span)>
        </glamorous(td)>,
        <glamorous(td)
          align="right"
        >
          <glamorous(button)
            active={false}
            onClick={[Function]}
          >
            W
          </glamorous(button)>
        </glamorous(td)>,
        <glamorous(td)>
          <glamorous(button)
            active={false}
            onClick={[Function]}
          >
            D
          </glamorous(button)>
        </glamorous(td)>,
        <glamorous(td)
          align="left"
        >
          <glamorous(button)
            active={false}
            onClick={[Function]}
          >
            W
          </glamorous(button)>
        </glamorous(td)>,
        <glamorous(td)>
          <glamorous(img)
            alt="Liverpool"
            draw={false}
            lose={false}
            src="https://www.thesun.co.uk/wp-content/uploads/2016/04/Liverpool.png?strip=all&w=67"
          />
          <glamorous(span)
            draw={false}
            lose={false}
          >
            Liverpool
          </glamorous(span)>
        </glamorous(td)>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <glamorous(img)
              alt="C Palace"
              draw={false}
              lose={false}
              src="https://www.thesun.co.uk/wp-content/uploads/2016/04/CPFC-Primary-Badge.png?strip=all&w=72"
            />,
            <glamorous(span)
              draw={false}
              lose={false}
            >
              C Palace
            </glamorous(span)>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "alt": "C Palace",
              "draw": false,
              "lose": false,
              "src": "https://www.thesun.co.uk/wp-content/uploads/2016/04/CPFC-Primary-Badge.png?strip=all&w=72",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "C Palace",
              "draw": false,
              "lose": false,
            },
            "ref": null,
            "rendered": "C Palace",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "align": "right",
          "children": <glamorous(button)
            active={false}
            onClick={[Function]}
          >
            W
          </glamorous(button)>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "active": false,
            "children": "W",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "W",
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <glamorous(button)
            active={false}
            onClick={[Function]}
          >
            D
          </glamorous(button)>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "active": false,
            "children": "D",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "D",
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "align": "left",
          "children": <glamorous(button)
            active={false}
            onClick={[Function]}
          >
            W
          </glamorous(button)>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "active": false,
            "children": "W",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "W",
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <glamorous(img)
              alt="Liverpool"
              draw={false}
              lose={false}
              src="https://www.thesun.co.uk/wp-content/uploads/2016/04/Liverpool.png?strip=all&w=67"
            />,
            <glamorous(span)
              draw={false}
              lose={false}
            >
              Liverpool
            </glamorous(span)>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "alt": "Liverpool",
              "draw": false,
              "lose": false,
              "src": "https://www.thesun.co.uk/wp-content/uploads/2016/04/Liverpool.png?strip=all&w=67",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Liverpool",
              "draw": false,
              "lose": false,
            },
            "ref": null,
            "rendered": "Liverpool",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <glamorous(td)>
            <glamorous(img)
              alt="C Palace"
              draw={false}
              lose={false}
              src="https://www.thesun.co.uk/wp-content/uploads/2016/04/CPFC-Primary-Badge.png?strip=all&w=72"
            />
            <glamorous(span)
              draw={false}
              lose={false}
            >
              C Palace
            </glamorous(span)>
          </glamorous(td)>,
          <glamorous(td)
            align="right"
          >
            <glamorous(button)
              active={false}
              onClick={[Function]}
            >
              W
            </glamorous(button)>
          </glamorous(td)>,
          <glamorous(td)>
            <glamorous(button)
              active={false}
              onClick={[Function]}
            >
              D
            </glamorous(button)>
          </glamorous(td)>,
          <glamorous(td)
            align="left"
          >
            <glamorous(button)
              active={false}
              onClick={[Function]}
            >
              W
            </glamorous(button)>
          </glamorous(td)>,
          <glamorous(td)>
            <glamorous(img)
              alt="Liverpool"
              draw={false}
              lose={false}
              src="https://www.thesun.co.uk/wp-content/uploads/2016/04/Liverpool.png?strip=all&w=67"
            />
            <glamorous(span)
              draw={false}
              lose={false}
            >
              Liverpool
            </glamorous(span)>
          </glamorous(td)>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <glamorous(img)
                alt="C Palace"
                draw={false}
                lose={false}
                src="https://www.thesun.co.uk/wp-content/uploads/2016/04/CPFC-Primary-Badge.png?strip=all&w=72"
              />,
              <glamorous(span)
                draw={false}
                lose={false}
              >
                C Palace
              </glamorous(span)>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "alt": "C Palace",
                "draw": false,
                "lose": false,
                "src": "https://www.thesun.co.uk/wp-content/uploads/2016/04/CPFC-Primary-Badge.png?strip=all&w=72",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "C Palace",
                "draw": false,
                "lose": false,
              },
              "ref": null,
              "rendered": "C Palace",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "align": "right",
            "children": <glamorous(button)
              active={false}
              onClick={[Function]}
            >
              W
            </glamorous(button)>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "active": false,
              "children": "W",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "W",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <glamorous(button)
              active={false}
              onClick={[Function]}
            >
              D
            </glamorous(button)>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "active": false,
              "children": "D",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "D",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "align": "left",
            "children": <glamorous(button)
              active={false}
              onClick={[Function]}
            >
              W
            </glamorous(button)>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "active": false,
              "children": "W",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "W",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <glamorous(img)
                alt="Liverpool"
                draw={false}
                lose={false}
                src="https://www.thesun.co.uk/wp-content/uploads/2016/04/Liverpool.png?strip=all&w=67"
              />,
              <glamorous(span)
                draw={false}
                lose={false}
              >
                Liverpool
              </glamorous(span)>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "alt": "Liverpool",
                "draw": false,
                "lose": false,
                "src": "https://www.thesun.co.uk/wp-content/uploads/2016/04/Liverpool.png?strip=all&w=67",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Liverpool",
                "draw": false,
                "lose": false,
              },
              "ref": null,
              "rendered": "Liverpool",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
